query/src/sql/planner/format/indent_format.rs

```diff
@@ -37,4 +37,25 @@ where T: Display
         }
         Ok(())
     }
+
+    /// format TreeNode in a pretty way
+    pub fn format_pretty(&self) -> Result<String> {
+        let mut buf = String::new();
+        self.format_pretty_impl(0, false, &mut buf)?;
+        Ok(buf)
+    }
+
+    fn format_pretty_impl(&self, indent: usize, is_last: bool, f: &mut String) -> Result<()> {
+        if indent == 0 {
+            writeln!(f, "{}", &self.payload).unwrap()
+        } else if is_last {
+            writeln!(f, "{}└── {}", "│   ".repeat(indent - 1), &self.payload).unwrap()
+        } else {
+            writeln!(f, "{}├── {}", "│   ".repeat(indent - 1), &self.payload).unwrap()
+        }
+        for (i, child) in self.children.iter().enumerate() {
+            child.format_pretty_impl(indent + 1, i == self.children.len() - 1, f)?;
+        }
+        Ok(())
+    }
```

Comment:
I think it's better to create a new file `pretty_format.rs` for this.

Reply:
> I think it's better to create a new file `pretty_format.rs` for this.

fixed it😄